src/DataOnev2/KNB.jl

```diff
@@ -0,0 +1,19 @@
+abstract type KNB <: DataOnev2 end
+
+base_url(::KNB) = "https://knb.ecoinformatics.org/knb/d1/mn/v2/"
```

Comment:
You want to define:
`object_url(repo) = base_url(repo) * "object/"`
or similar 

Then remove a bunch of string literals from your code:
`"https://knb.ecoinformatics.org/knb/d1/mn/v2/object/"`

This follows the DRY principle https://en.wikipedia.org/wiki/Don%27t_repeat_yourself